simulator/Publisher.ts

```diff
@@ -17,42 +17,39 @@ export class Publisher implements IPublisher {
         "ROUND_START"
     ];
 
-    constructor() {
+    constructor(strategies: IPublishStrategy[]) {
 
-        this.setAllMessagesEnabled(true);
+        this.publishStrategies = strategies;
     }
 
-    public setPublishProvider(publishProvider: IPublishProvider) {
+    public setPublishProvider(publishProvider: IPublishProvider): void {
 
         this.publishProvider = publishProvider;
     }
 
-    public publish(message: IMessage): void {
-
-        if (!this.publishProvider) {
-            return;
-        }
-
-        if (!this.typeEnabled[message.type]) {
-            return;
-        }
+    public queue(message: IMessage): void {
 
-        this.publishProvider.publishSync(
-            this.typeDictionary[message.type],
-            message.payload
-        )
+        this.publishStrategies[message.type].queue(message);
     }
 
-    public setAllMessagesEnabled(enabled: boolean): void {
+    public publish(): void {
 
-        this.typeEnabled = [];
-        for (var i = 0; i < MessageType.Count; i++) {
-            this.typeEnabled.push(enabled);
+        if (!this.publishProvider) {
+            return;
         }
-    }
 
-    public setMessageTypeEnabled(type: MessageType, enabled: boolean): void {
+        this.publishStrategies
+            .forEach(s => {
+                var message = s.dequeue();
+
+                if (!message) {
+                    return;
+                }
 
-        this.typeEnabled[type] = enabled;
+                this.publishProvider.publishSync(
+                    this.typeDictionary[message.type],
```

Comment:
Should this just be `message`?

Reply:
The `typeDictionary` bit turns enums into strings e.g. 1 becomes CORE_ACCESS etc.